src/playlist-selectors.js

```diff
@@ -124,27 +146,32 @@ const simpleSelector = function(master, bandwidth, width, height) {
   let resolutionPlusOneSmallest = [];
   let resolutionBestVariantList = [];
 
-  stableSort(sortedPlaylists, comparePlaylistBandwidth);
+  stableSort(sortedPlaylists, comparePlaylistBandwidth.bind(null, useAverageBandwidth));
 
   // filter out any playlists that have been excluded due to
   // incompatible configurations or playback errors
   sortedPlaylists = sortedPlaylists.filter(Playlist.isEnabled);
   // filter out any variant that has greater effective bitrate
   // than the current estimated bandwidth
   bandwidthPlaylists = sortedPlaylists.filter(function(elem) {
-    return elem.attributes &&
-           elem.attributes.BANDWIDTH &&
-           elem.attributes.BANDWIDTH * Config.BANDWIDTH_VARIANCE < bandwidth;
+    const elemBandwidth = getBandwidth(elem, useAverageBandwidth);
```

Comment:
I believe there should be more places that it makes a difference. For instance, `comparePlaylistResolution` will use BANDWIDTH vs AVERAGE-BANDWIDTH if there is a matching resolution. In addition, it keeps consistency with what bandwidth we are using for comparisons (versus overwriting a playlist's attribute/maintaining a pseudo playlist with altered values), and also combines a lot of the standard guards against missing/empting `attributes` and `BANDWIDTH` properties.